frontend/lib/ui/artefact_dialog/test_execution_pop_over.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:yaru_widgets/yaru_widgets.dart';
+
+import '../../models/test_execution.dart';
+import '../../providers/artefact_builds.dart';
+import '../spacing.dart';
+
+class TestExecutionPopOver extends ConsumerStatefulWidget {
+  const TestExecutionPopOver({
+    super.key,
+    required this.testExecution,
+    required this.artefactId,
+  });
+
+  final TestExecution testExecution;
+  final int artefactId;
+
+  @override
+  TestExecutionPopOverState createState() => TestExecutionPopOverState();
+}
+
+class TestExecutionPopOverState extends ConsumerState<TestExecutionPopOver> {
+  TextEditingController reviewCommentController = TextEditingController();
+  List<TestExecutionReviewDecision> reviewDecisions = [];
+
+  bool get _canApprove {
+    return !reviewDecisions.contains(TestExecutionReviewDecision.rejected);
+  }
+
+  bool get _canReject {
+    return reviewDecisions.isEmpty ||
+        reviewDecisions.contains(TestExecutionReviewDecision.rejected);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    reviewCommentController.text = widget.testExecution.reviewComment;
+    reviewDecisions = List.from(widget.testExecution.reviewDecision);
+  }
+
+  @override
+  void dispose() {
+    reviewCommentController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      padding: const EdgeInsets.all(12),
+      children: [
+        Text(
+          'Select new review status:',
+          style: Theme.of(context).textTheme.titleLarge,
+        ),
+        const SizedBox(height: Spacing.level3),
+        Column(
+          children: TestExecutionReviewDecision.values
+              .map(
+                (e) => YaruCheckboxListTile(
+                  value: reviewDecisions.contains(e),
+                  onChanged: ((e == TestExecutionReviewDecision.rejected &&
+                              (_canReject)) ||
+                          (e != TestExecutionReviewDecision.rejected &&
+                              _canApprove))
+                      ? (bool? value) {
+                          setState(() {
+                            if (reviewDecisions.contains(e)) {
+                              reviewDecisions.remove(e);
+                            } else {
+                              reviewDecisions.add(e);
+                            }
+                          });
+                        }
+                      : null,
```

Comment:
can you pull `onChanged` as a separate helper method/function as it's a bit hard to follow this way

Reply:
I now extracted this in a separate function.